internal/controllers/synthesis/lifecycle.go

```diff
@@ -115,13 +106,17 @@ func (c *podLifecycleController) Reconcile(ctx context.Context, req ctrl.Request
 			return ctrl.Result{}, nil
 		}
 
+		// TODO: Is it possible to delete before the first synthesis?
```

Comment:
When is the finalizer set? Technically if you delete right after creation, yes, right?

Reply:
I was thinking about if a composition is deleted between the time its finalizer is added and the time its first synthesis has succeeded. But it should be okay since we continue on to remove the finalizer in cases where current state is nil. Will remove this note